commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -63,25 +56,19 @@ public boolean equals(Object o) {
    }
 
    public boolean equalsWrappedBytes(WrappedBytes other) {
-      int length = getLength();
-      if (other.getLength() != length) return false;
-      if (other.hashCode() != hashCode()) return false;
-      for (int i = 0; i < length; ++i) {
-         if (getByte(i) != other.getByte(i)) return false;
-      }
-      return true;
+      //Array.equals already checks the length and hashcode is lazy now
+      return Arrays.equals(bytes, other.getBytes());
```

Comment:
Unfortunately we can't do this.  There is no guarantee that getBytes will return a byte[].  For example when using off heap this will always be null.